### ISSUE TITLE
Feature Request - rework filesystem - speed increase

### ISSUE THREAD

**Issue by lewisdonofrio** [CypherpunkArmory/UserLAnd#721]
# Describe the problem your feature request solves.
How do I get the new speed without having to reinstall my daily used ubuntu 18.04 lts arm64 setup?

# Describe the solution you'd like 
You stated you had a script that could do this but have yet to release it.  I'm looking to help test this script in whatever form it is at.

# Describe alternatives you've considered
need to get my speed to be as fast as possible.  Plese post the scipt you used to keep data and rework filesystem.

# Additional context
If you need to add a backup install and rework/reformat filesystem for huge speed increase, will wipe in a week and reload because updates take forever currently.

built with http://www.tinyurl.com/donofriodexnote8

**Comment by corbinlc** [comment]
Right.  I didn't get back, because I realized if we had a way to backup and restore, that would solve things as it would use the new version of PRoot for the filesystem when you restored it.  But, now we do support backup and restore.  So, here is what you need to do...
1) On the filesytem page, long press the filesystem you want to backup and select that you want to back it up.
2) Then, on the filesystem page, click the + button and describe a new filesystem with the same username and password and distro that you used before.
3) Then on the sessions page, describe a new ssh or vnc session that uses that new filesystem.
4) Click on that session (notice that restore does not work with the "app" flow right now, only the "session" flow).  This will regenerate your filesystem and launch you into it, but with the new version of PRoot.
Hopefully, you have enough room to not have to delete the old filesystem before performing restore so you can try it out before axing it.

**Comment by lewisdonofrio** [comment]
#1 guessing you mean "export?"

**Comment by lewisdonofrio** [comment]
help....how do I fix this?
![Screenshot_20190326-103759_UserLAnd](https://user-images.githubusercontent.com/3161117/55006016-6bedd600-4fb3-11e9-84e7-a85025bfcfc6.jpg)

**Comment by lewisdonofrio** [comment]
is this the file/directory to delete to try another export/backup?
![Screenshot_20190326-104147_My Files](https://user-images.githubusercontent.com/3161117/55006972-fbe04f80-4fb4-11e9-8c82-abd4cc00f441.jpg)

**Comment by MatthewTighe** [comment]
Yeah, it is. @corbinlc left out a part of step 2.

Navigate to the filesystem edit page by clicking the + while on the filesystems page. The new filesystem must use the same username, password, and distribution (Ubuntu, in your case) as the exported filesystem. Scroll down to "advanced options" and you will see a button to import the filesystem you exported. Click the save icon. The first time you start a session on that filesystem, it should extracted the backup.

**Comment by lewisdonofrio** [comment]
how do I get the 'import' button to show up so far all I see is:
![Screenshot_20190326-121935_UserLAnd](https://user-images.githubusercontent.com/3161117/55016914-15d65e00-4fc6-11e9-9feb-2a76e0e93c7f.jpg)

**Comment by corbinlc** [comment]
@lewisdonofrio  there is no "import" button.  You select the backup file to import here, but it doesn't get used until you go to launch the session for the first time.

**Comment by lewisdonofrio** [comment]
I opened the new session but nothing get's restored, I mean how do I know it's being restored....just feeling almost done but missing the show type of feeling....I have the tgz file I have a the new filesystem with compressed file selected and when I start the sesson I get the user login prompt then it drops me in like a new session to the prompt ans such...

**Comment by corbinlc** [comment]
If you created a session that is targeting the new filesystem you specified (you have to tell it which filesystem the session is targeting), then it should show you that it is extracting a new filesystem the first time you launch it.  Does it not?

**Comment by lewisdonofrio** [comment]
nope not as of yet......

**Comment by lewisdonofrio** [comment]
just says "starting service." nothing about, please wait 15 min while restore is in progress....or with new proot it should be 3 min....

**Comment by lewisdonofrio** [comment]
ok I created the 1st attempt at new proot filesystem, I forgot to assign it a tgz file, so I went back and edited it and tried to save but I don't think it updated or something.  my 2nd attempt, removed new session and new filesystem, create new filesystem, set type to ubuntu select import tgz and save and poof, it is not in the list of filesystems.  try again 3rd time, poof still no filesystem like when I made it without assigning an import tgz.....

**Comment by lewisdonofrio** [comment]
now I'm stuck cannot create a new filesystem ;(  I mean my normal filesystem works but that is the debug speed proot....it's ok just want full speed......so I'll wait... thank you all for your help so far...

**Comment by corbinlc** [comment]
Please see if this matches what you are doing..

Part 1: Launch into an existing file system and create a test file so we can see if it gets restored later.  Then go to the filesystem page and start the backup (cut the video off because it takes a bit to make a backup)
https://www.dropbox.com/s/pruihmr366yy74l/backup-part1.mp4?dl=0

Part 2: Create a filesystem from the back up.  Then create a new session which uses that backup.  Then launch that session for the first time and see that it is in fact extracting a new filesystem.
https://www.dropbox.com/s/7906y2f8usmcdve/backup-part2.mp4?dl=0

Part 3: Launch into the new filesystem to show that the test file is there and things do appear to be restored.
https://www.dropbox.com/s/ytet7t6gf7mnido/backup-part3.mp4?dl=0

What is different about what you are doing?

**Comment by lewisdonofrio** [comment]
I followed part1 fine, part2 is what I was doing, and I did see something about files being moved but my goes-16.sh was not there I believe along with ansible and remmina and etc.  now it seems I have another issue, userland filesystem is not saving another ubuntu while one already exists it seems, cause I tried and tired and tired and just now when I switched to kail for filesystem and saved it saved and showed up an a selection, how do I get a second ubuntu like the first time I tired this?

**Comment by lewisdonofrio** [comment]
eek now I cannot select os say debian and restore file for ubuntu it doesn't save that filesystem attempt...did you feel like doing a teamviewer of my install/setup and see if you can at least see what I'm seeing...(or worse case what is that screen recorders app you used just then?)

**Comment by corbinlc** [comment]
I am not sure why this is happening.  The screen recording I used was from inside Android Studio.  You can do screen shots and captures from there.  It just uses ADB, so maybe you could do it without all of Android Studio, but just installing Android Studio is probably the easiest route.  

This is also a fun tool.  I haven't used it for screen capture, but others have... https://github.com/Genymobile/scrcpy

**Comment by lewisdonofrio** [comment]
teamviewer works when you have free time.....or I might just wipe out my work to date and just reinstall the ubuntu app session.....was so hoping to hold off till 19.04 is released in a week.....so if you get time I'm game to debug this (heck I even turned debugging on just to see if this filesystem creating error is something trackable....)

**Comment by corbinlc** [comment]
@MatthewTighe any new thoughts on this?  Specifically, the last couple of issues @lewisdonofrio was running into.

**Comment by corbinlc** [comment]
Closing as part of a repository cleanup. If this issue is still relevant, feel free to reopen it or file a fresh one.
